tests/test_pumptrap_calibration.py

```diff
@@ -0,0 +1,274 @@
+    data_filenames = sorted(glob.glob(os.path.join(test_data_dir, "*.fits")))
+    pump_trap_dataset = Dataset(data_filenames)
+
+    #Parse the first three characters of each filename into a temperature
+    # temps = [int(os.path.basename(f)[:3]) for f in data_filenames]
+    # npumps = [int(os.path.basename(f).split("_")[5]) for f in data_filenames]
+    # scheme = [int(os.path.basename(f).split("_")[2]) for f in data_filenames]
+    #Hack in some missing header parameters. 
+    arrtype = 'SCI'
+    # em_gain = 10 #The default in generate_test_data.py
+
+    for j,frame in enumerate(pump_trap_dataset):
+        frame.ext_hdr['ARRTYPE'] = arrtype
+        # frame.ext_hdr['CMDGAIN'] = em_gain
+        # frame.ext_hdr['EXCAMT'] = temps.pop(0)
+
+        #Get the scheme the the "scheme" list. For the current scheme set the header keyword TPSCHEM* 
+        # equal to the npumps for this filename, where * is equal to the scheme. For the other schemes (up to * =4) set TPSCHEM* =0.
+        # for i in range(1, 5):
+        #     if scheme[j] == i:
+        #         frame.ext_hdr['TPSCHEM' + str(i)] = npumps.pop(0)
+        #     else:
+        #         frame.ext_hdr['TPSCHEM' + str(i)] = 0
+
+        #Get the phase time from the filename: its between the string "phasetime" and the ".fits" extension at the end
+        # phase_time = os.path.basename(data_filenames[j]).split("phasetime")[1].split(".fits")[0]
+        # frame.ext_hdr['TPTAU'] = float(phase_time)
+
+    #Run the bias subtraction
+    # #TODO Figure out which detector regions to pass in here. 
+    # arrtype = pump_trap_dataset[0].ext_hdr['ARRTYPE']
+
+    #Detector regions for the smaller pump_trap_data - taken from metadata_test.yaml
+    detector_regions = {
+        arrtype : {
+            'frame_rows': 120, 
+            'frame_cols': 220,
+            'image': {
+                'rows': 104,
+                'cols': 105,
+                'r0c0': [2, 108]
+                },
+            'prescan_reliable': {
+                'rows': 120,
+                'cols': 108,
+                'r0c0': [0, 0],
+                'col_start': 0, #10
+                'col_end': 108, #100
+            },        
+            'prescan': {
+                'rows': 120,
+                'cols': 108,
+                'r0c0': [0, 0],
+                'col_start': 0, #10
+                'col_end': 108, #100
+            }, 
+            'serial_overscan': {
+                'rows': 120,
+                'cols': 5,
+                'r0c0': [0, 215]
+            },
+            'parallel_overscan': {
+                'rows': 14,
+                'cols': 107,
+                'r0c0': [106, 108]
+            }
+            },
+        }
+
+    #Subtract the prescane Bias
+    bias_subbed_dataset = prescan_biassub(pump_trap_dataset, detector_regions=detector_regions,use_imaging_area=True)
+
+    bias_subbed_dataset[0].save(filedir = "./", filename="bias_subbed_dataset_file0.fits")
```

Comment:
does this file and the emgain save file need to happen? It's not clear they are ever used.

If they are used, the outputdir should not be `./`. It should be defined relative to this file, so that it is always saved to the same location.

Reply:
I don't really know why I was doing this...